DataFormats/CTPPSReco/interface/CTPPSPixelCluster.h

```diff
@@ -0,0 +1,134 @@
+#ifndef DataFormats_CTPPS_CTPPSPixelCluster_h
+#define DataFormats_CTPPS_CTPPSPixelCluster_h
+
+/*
+  \class CTPPSPixelCluster 
+  \brief CTPPSPixelCluster stores the information of CTPPS Tracker clusters of 3D pixels
+ Author: F.Ferro - INFN Genova - 2016
+*/
+
+
+#include <vector>
+#include <cstdint>
+#include <cassert>
+
+
+class CTPPSPixelCluster {
+
+public:
+
+  CTPPSPixelCluster() {}
+  static constexpr uint8_t MAXSPAN=255;
+  static constexpr uint8_t MAXCOL=155; 
+  static constexpr uint8_t MAXROW=159; 
+
+  
+CTPPSPixelCluster(uint16_t isize, uint16_t * adcs,         
+		  uint8_t const * rowpos,  uint8_t const * colpos, 
+		  uint8_t const  rowmin,  uint8_t const  colmin) :
+  thePixelOffset(2*isize), // the pixel offset is the pixel position inside the cluster wrt rowmin (even positions) and colmin (odd positions)
+    thePixelADC(adcs, adcs+isize)   
+    {
+
+      uint8_t maxCol = 0;
+      uint8_t maxRow = 0;
+      for (unsigned int i=0; i!=isize; ++i) {
+	uint8_t rowoffset = rowpos[i]-rowmin;
```

Comment:
I see the answer - it is promised - but it would be safer and more efficient to move the calculation of rowmin to this function, where it could be done once per cluster instead of numerous times.